components/raftstore/src/store/fsm/peer.rs

```diff
@@ -1529,6 +1531,23 @@ impl<'a, T: Transport, C: PdClient> PeerFsmDelegate<'a, T, C> {
         // Check if snapshot file exists.
         self.ctx.snap_mgr.get_snapshot_for_applying(&key)?;
 
+        // WARNING: The checking code must be above this line.
+        // Now all checking passed.
+
+        if self.ctx.cfg.dev_assert && self.fsm.peer.create_peer_kind == CreatePeerKind::FromCreating
+        {
+            // If the region is not initialized and snapshot checking has passed, the split flag must be false.
+            // The split process only change this flag to true if it's exactly the same peer,
+            // if so, this peer can't pass the previous range check.
+            // The condition of passing the range check is this region must merge the other regions and then split.
+            // It's impossible because this peer does not exist which does not satisfy the merge condition, i.e.
+            // all target peer must exist during merging.
+            let pending_create_peers = self.ctx.pending_create_peers.lock().unwrap();
+            assert_eq!(
```

Comment:
Better use panic to include useful information like region id and peer id.

Reply:
Done.